packages/aws-client/tests/test-services.js

```diff
@@ -362,15 +363,23 @@ test('sqs() service defaults to localstack in test mode', async (t) => {
   );
 });
 
-test('sts() service defaults to localstack in test mode', (t) => {
+test('sts() service defaults to localstack in test mode', async (t) => {
   const sts = services.sts();
   const {
     credentials,
     endpoint,
-  } = localStackAwsClientOptions(AWS.STS);
+  } = localStackAwsClientOptions(STS);
   t.deepEqual(
-    sts.config.credentials,
+    await sts.config.credentials(),
     credentials
   );
-  t.is(sts.config.endpoint, endpoint);
+  const serviceConfigEndpoint = await sts.config.endpoint();
+  const localEndpoint = new URL(endpoint);
+  t.like(
```

Comment:
Nit: we should generally be avoiding like unless there's a good reason (non-deterministic or absurdly large generated values vs the fixture) - it probably doesn't serve anything here to worry about it as the goal is to validate it's switched to localstack, not validate the configuration. 

All of the above said, non-blocking, but consider removing the port from the test given we really only care that it's targeting localhost.

Reply:
I made this change based on previous SDK v3 PRs from other team members. Thanks for the review.